src/snitch_section.cpp

```diff
@@ -89,6 +117,11 @@ section_entry_checker::operator bool() {
         (level.current_section_id == level.previous_section_id &&
          state.sections.depth < state.sections.levels.size())) {
 
+        // First time entering this section, emit the section start event
+        if (level.current_section_id == level.previous_section_id + 1 &&
+            level.current_section_id != level.previous_section_id) {
```

Comment:
I don't think this is right; the second part of the expression after the `&&` will always evaluate to `true` if the first part also evaluates to `true`. Perhaps this simply needs to be removed:
```suggestion
        if (level.current_section_id == level.previous_section_id + 1) {
```